apps/src/code-studio/progress.js

```diff
@@ -336,28 +337,33 @@ function initializeStoreWithSections(store, scriptData) {
     return;
   }
 
-  // If we have a selected section, merge it with the minimal data in the
-  // `sections` array before storing in redux.
   const currentSection = scriptData.section;
-  if (currentSection) {
-    const idx = sections.findIndex(section => section.id === currentSection.id);
-    if (idx >= 0) {
-      sections[idx] = {
-        ...sections[idx],
-        ...currentSection
-      };
-    }
+  if (!currentSection) {
+    // If we don't have a selected section, simply set sections and we're done.
+    store.dispatch(setSections(sections));
```

Comment:
This looks fine, but just curious, why couldn't we just move `store.dispatch` above this if statement (and remove it from below).  Do we have to call setSections after the findIndex?

Reply:
maybe i should explain this better in the comments, but this is a bit of a hack. `scriptData.sections` is an array of sparsely-populated section objects, whereas `scripData.section` is a fully-populated object with more data that we need in redux, but only for the current section. so if we have a current section, we merge that into the `sections` array before storing into redux. so we need to dispatch `setSection` after the merge but before `selectSection`

Reply:
Ohhh, I see.  Hmmm, yeah, that is a bit confusing.  Without looking at the rest of the code and understanding its nuances, I would say that it'd be better to have all of the items in sections be minimal and then keep a separate state variable currentSection that includes more details.

Anyway, this PR looks good, thanks for taking the time to explain this to me.

Reply:
i agree about storing it separately, but that whole redux tree needs cleanup so i decided this temporary hack was more appropriate for this scope.